### DISCUSSION TITLE
producing messages outside of a consumer

### DISCUSSION THREAD

**Discussion by edwja** [zendesk/racecar#279] [Q&A] [answered]
We are in the process of introducing (Heroku) Kafka into our ecosystem (several Ruby apps in Heroku) to get all of the Goodness of message-driven architecture.  We put together a POC where we produce messages in once service and consume them in another.  We plan to expand this to many messages produced and consumed across many apps.

We are now in the process of deciding on a higher-level message processing framework that plays well with Rails.  In Racecar, what is the recommended way to produce messages when they are not in response to consuming an event?  It seems that any chain of messages has to start with an initially produced message.

**Comment by dasch** [comment, marked as answer]
I recommend using [DeliveryBoy](https://github.com/zendesk/delivery_boy) in Rails applications when producing messages outside of a consumer context.
